ui-tests/cypress/support/pageObjects/content/types/AddPage.js

```diff
@@ -0,0 +1,47 @@
+import {TEST_ID_KEY, htmlElements, WebElement} from '../../WebElement';
+import Content from '../../app/Content.js';
+import AppPage from '../../app/AppPage.js';
+import TypesPage from './TypesPage.js';
+import EditPage from './EditPage';
+
+export default class AddPage extends Content {
+
+  nameInput = `${htmlElements.input}[name=name]`;
+  codeInput = `${htmlElements.input}[name=code]`;
+
+  getNameInput() {
+    return this.getContents()
+               .find(this.nameInput);
+  }
+
+  getCodeInput() {
+    return this.getContents()
+               .find(this.codeInput);
+  }
+
+  getSaveButton() {
+    return this.getContents()
+               .find(htmlElements.button)
+               .contains('Save');
```

Comment:
Avoid using the text to identify elements, that is related to UX/UI requirements is not a structural information

Reply:
Odd though that Cypress uses `contains()` in their example repo to select buttons. But yeah after going through their best practices page, I've seen that they think using `data-*` attributes are better. Thanks!